### ISSUE TITLE
Bug: no-magic-numbers Doesn't apply ignore to type definitions

### ISSUE THREAD

**Issue by anthony-unicare** [typescript-eslint/typescript-eslint#9161]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=KYDwDg9gTgLgBDAnmYcDyA7YaoBUDuEcAvHAIxwA%2BcATANxA&eslintrc=N4KABGBEAOA2CuBzAlgO0gLjAbXBKAAgC4Ce0ApgM4DGATstEQLRWxpGR4C6ANHpLXiwqmMKHyFSFGvUYtKbVEQD0qAPZMAtgEMU1JqniaARuVqVR2SAHdttdDzGRkidbXKWAjI4BMXAL5ceP4g-kA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
export type OneOrTwo = 1 | 2;
```

### ESLint Config

```javascript
{
  "plugin": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/no-magic-numbers": ["warn", {"ignore": [1, 2]}]
  }
}
```

### Expected Result

I expected that because the numeric literals 1 and 2 are in the "ignore" setting of the "no-magic-numbers" rule, they would not be reported as magic numbers.

### Actual Result

The numeric literals 1 and 2 are reported as magic numbers, despite the rule being configured to ignore them.

### Additional Info

Here's a patch:

```
diff --git a/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-magic-numbers.js b/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-magic-numbers.js
index 07b7055..c5b6014 100644
--- a/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-magic-numbers.js
+++ b/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-magic-numbers.js
@@ -25,6 +25,17 @@ Array.isArray(baseRule.meta.schema)
         },
     },
 });
+/**
+ * Convert the value to bigint if it's a string. Otherwise return the value as-is.
+ * @param {bigint|number|string} x The value to normalize.
+ * @returns {bigint|number} The normalized value.
+ */
+function normalizeIgnoreValue(x) {
+    if (typeof x === "string") {
+        return BigInt(x.slice(0, -1));
+    }
+    return x;
+}
 exports.default = (0, util_1.createRule)({
     name: 'no-magic-numbers',
     meta: {
@@ -49,6 +60,7 @@ exports.default = (0, util_1.createRule)({
     ],
     create(context, [options]) {
         const rules = baseRule.create(context);
+        const ignore = new Set((options.ignore || []).map(normalizeIgnoreValue));
         return {
             Literal(node) {
                 // If it’s not a numeric literal we’re not interested
@@ -76,6 +88,10 @@ exports.default = (0, util_1.createRule)({
                 else if (isParentTSReadonlyPropertyDefinition(node)) {
                     isAllowed = options.ignoreReadonlyClassProperties === true;
                 }
+                // Check if the value is allowed
+                if (ignore.has(node.value)) {
+                    isAllowed = true;
+                }
                 // If we’ve hit a case where the ignore option is true we can return now
                 if (isAllowed === true) {
                     return;
```

**Comment by bradzacher** [comment]
Looks like this regression landed over 2y ago in https://github.com/typescript-eslint/typescript-eslint/pull/4789.

Originally the rule would defer to the base implementation (which would respect `ignore`) but in that PR the logic was refactored and that logic was lost.

**Comment by Vinccool96** [comment]
> The fix would be to fix the logic so that we fall back to the base rule logic for types.

I'll do it, shouldn't take me long. I'll also add this case to the tests, so that it doesn't become an issue again.

**Comment by Vinccool96** [comment]
After adding some tests, I realized that, even before #4789, magic numbers that should have been ignored with the option `[{ ignoreReadonlyClassProperties: false, ignore : [<The>, <Readonly>, <Class>, <Properties>] }]` would have been caught. I'll need a bit longer to fix it.

Just goes to prove that the rule is extremely rarely used

**Comment by Josh-Cena** [comment]
> Hilarious that it hasn't been detected in 2y though. Shows you how little the rule is used.

#4860 sent 2y ago: sitting in the corner

**Comment by Vinccool96** [comment]
> > Hilarious that it hasn't been detected in 2y though. Shows you how little the rule is used.
> 
> #4860 sent 2y ago: sitting in the corner

Well now I feel bad

**Comment by Vinccool96** [comment]
> @Vinccool96, is the patch I provided in my initial report acceptable? If so, could we please get it merged?

While working on it, I realized that the problem was a bit bigger than what you brought to our attention. Only merging your patch would be equivalent to changing the oil when you need to replace the pistons.
